src/moonlink/src/storage/iceberg/tests.rs

```diff
@@ -976,11 +975,11 @@ async fn test_index_merge_and_create_snapshot_impl(iceberg_table_config: Iceberg
 
     // Attempt index merge and flush to iceberg table.
     create_mooncake_and_iceberg_snapshot_for_index_merge_for_test(&mut table, &mut notify_rx).await;
-
+    let object_storage_cache = create_test_object_storage_cache(&cache_temp_dir);
     // Create a new iceberg table manager and check states.
     let mut iceberg_table_manager_for_recovery = IcebergTableManager::new(
         mooncake_table_metadata.clone(),
-        ObjectStorageCache::default_for_test(&cache_temp_dir), // Use separate cache for each table.
+        object_storage_cache, // Use separate cache for each table.
```

Comment:
I strongly prefer
```suggestion
        create_test_object_storage_cache(&cache_temp_dir), // Use separate cache for each table.
```
which makes it easier to tell it's 100% a new object storage cache.

The reason I'm extremely picky is, there's background threads accessing cache, which makes test non-deterministically passing or failing.